Common/cpp/LayoutAnimations/LayoutAnimationsManager.cpp

```diff
@@ -123,9 +139,27 @@ std::unordered_map<int, std::shared_ptr<Shareable>>
       return layoutAnimations_;
     case SHARED_ELEMENT_TRANSITION:
       return sharedTransitionAnimations_;
+    case SHARED_ELEMENT_TRANSITION_PROGRESS:
+      return sharedTransitionProgressAnimations_;
     default:
       assert(false);
   }
 }
 
+void LayoutAnimationsManager::updateSharedTransitionProgress(
+    const std::shared_ptr<JSRuntimeHelper> &runtimeHelper,
+    const int sourceViewTag,
+    const int targetViewTag,
+    const double progress) {
+  std::shared_ptr<Shareable> config;
+  {
+    auto lock = std::unique_lock<std::mutex>(animationsMutex_);
+    config =
+        getConfigsForType(SHARED_ELEMENT_TRANSITION_PROGRESS)[sourceViewTag];
+  }
+  jsi::Value progressUpdater = config->getJSValue(*runtimeHelper->uiRuntime());
+  runtimeHelper->runOnUIGuarded(progressUpdater, sourceViewTag, progress);
+  runtimeHelper->runOnUIGuarded(progressUpdater, targetViewTag, progress);
```

Comment:
Wonder why we sometimes pass source,target,progress and other times we only pass a single tag. This is perhaps not an error and certain flows are just set up this way but it feels conceptually inconsistent